File: kabbalah2.c
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 */

#include <assert.h>
#include <stdio.h>

// constraints
#define MAXN 1001
#define MAXM 1001

// input data
int N, M, i, j;
char mat[MAXN][MAXM];

int main() {
  //  uncomment the following lines if you want to read/write from files
  //  freopen("input.txt", "r", stdin);
  //  freopen("output.txt", "w", stdout);

  assert(2 == scanf("%d %d", &N, &M));
  for (i = 0; i < N; i++) {
    assert(1 == scanf("%s", mat[i]));
  }

  // insert your code here

  printf("%d\n", 4242);  // change 4242 with actual answer
  return 0;
}

File: kabbalah2.pas
{ constraints }
const
    MAXN = 1000;

{ input data }
var
    N, M, i : longint;
    mat     : array[0..MAXN-1] of string;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N, M);
    for i:=0 to N-1 do
        readln(mat[i]);

    { insert your code here }

    writeln(4242); { change 4242 with actual answer }
end.
